docs/rolling_average_api_working.md

```diff
@@ -0,0 +1,191 @@
+
+# Info dump on rolling average APIs #
+
+Rolling averages are currntly nasty to do with with timescaledb (user complaint https://news.ycombinator.com/item?id=27051005).  In our timeseries API we will eventually provide a function like
+```SQL
+moving_average(window => '30 minutes', slide => '5 minutes', data)
+```
+However, because set-returning aggregates cannot not exist in Postgres, this will not work outside of the timeseries API. Currently, doing rolling average properly requires windowed aggregates. In base SQL it is a real PITA because you have to do sum and count separately and then divide them yourself.
```

Comment:
We should probably also think about the continuous aggregate implications ie https://github.com/timescale/timescale-analytics/issues/138 in this document.